recipes/libpng/all/conanfile.py

```diff
@@ -51,31 +66,27 @@ def _patch(self):
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}',
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/$<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}')
 
-        if self.settings.build_type == 'Debug':
-            tools.replace_in_file(os.path.join(self._source_subfolder, 'libpng.pc.in'),
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@',
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@d')
-
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PNG_TESTS"] = "OFF"
-        cmake.definitions["PNG_SHARED"] = self.options.shared
-        cmake.definitions["PNG_STATIC"] = not self.options.shared
-        cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["PNG_TESTS"] = "OFF"
+        self._cmake.definitions["PNG_SHARED"] = self.options.shared
+        self._cmake.definitions["PNG_STATIC"] = not self.options.shared
+        self._cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
         if self.settings.os == "Emscripten":
-            cmake.definitions["PNG_BUILD_ZLIB"] = "ON"
-            cmake.definitions["M_LIBRARY"] = ""
-            cmake.definitions["ZLIB_LIBRARY"] = self.deps_cpp_info["zlib"].libs[0]
-            cmake.definitions["ZLIB_INCLUDE_DIR"] = self.deps_cpp_info["zlib"].include_paths[0]
-        if self.settings.os == "Macos":
-            if 'arm' in self.settings.arch:
-                cmake.definitions["PNG_ARM_NEON"] = "on"
-        if self.options.api_prefix:
-            cmake.definitions["PNG_PREFIX"] = self.options.api_prefix
-        if self.settings.os == "Macos" and self.settings.arch == "armv8":
-            cmake.definitions["CMAKE_SYSTEM_PROCESSOR"] = "aarch64"
-        cmake.configure()
-        return cmake
+            self._cmake.definitions["PNG_BUILD_ZLIB"] = "ON"
```

Comment:
Does this mean a vendored `zlib` is built by libpng when building for emscripten?

Reply:
I guess yes

Reply:
Looks like no zlib is used at all:
https://github.com/glennrp/libpng/blob/a37d4836519517bdce6cb9d956092321eca3e73b/CMakeLists.txt#L38-L45

Reply:
yes, maybe there was an issue with Emscripten in conan long time ago.
It's the same thing for M_LIBRARY, it's useless to force it empty value, it is overridden by upstream anyway.

Reply:
removed in https://github.com/conan-io/conan-center-index/pull/6285/commits/2ebb94fc64a3e849d0e7d491eb95c748f8ee22b6

Reply:
There is a pr on cci for emscripten at https://github.com/conan-io/conan-center-index/pull/6163
Once it gets merged, some recipes should be tested with it.